src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeEventDispatcher.cs

```diff
@@ -179,6 +187,11 @@ private unsafe void DispatchEventsToEventListeners()
                 // Wait for more events.
                 if (!m_stopDispatchTask)
                 {
+                    if (!eventsReceived)
```

Comment:
I'm loathe to tell you to change the factoring given the timeline for trying to get fixes in, but fwiw propagating the handle back to managed code seems like an unnecessarily complex interface boundary. What seemed simpler would be putting all the logic to block and/or sleep inside EventPipeBufferManager::GetNextEvent().
I'd suggest just adding a comment suggesting this could be refactored in the future.

Reply:
Comment added